pkg/deploy/generator/templates.go

```diff
@@ -158,6 +159,9 @@ func (g *generator) rpGlobalTemplate() *arm.Template {
 	t.Resources = append(t.Resources,
 		g.rpVersionStorageAccount()...,
 	)
+	t.Resources = append(t.Resources,
```

Comment:
These appends are getting out of hand as we are copying them around the place. We usually do pattern:
```
append(resource, resource1, resource2)
append(resource, compositeresources1...)
append(resource, compositeresources2...)
```
You are creating single resource here,
can you do the same:
```
t.Resources = append(t.Resources,
	g.rpBootDiagnosticsStorageAccount(),
)
t.Resources = append(t.Resources,
	g.acrRbac()...,
)
t.Resources = append(t.Resources,
	g.rpVersionStorageAccount()...,
)
```
where rpBootDiagnosticsStorageAccount returns single resource .

Reply:
Good :eye:. There used to be a storage container resource in here, and hence, the slice. I removed the storage container since the VM boot diagnostic extension automatically creates a storage container for each node in the vmss.